lib/html-composer.js

```diff
@@ -94,6 +96,12 @@ function applyOptions(options, res) {
 
     // can't assign to result, or we'd get a circular reference in refs.
     _.set(state, '_components', componentList);
+    _.set(state, '_componentSchemas', _.map(componentList, function (name) {
+      return {
+        name: name,
+        schema: schema.getSchema(files.getComponentPath(name))
+      };
+    }));
```

Comment:
Hmm. It does increase the size of the state object, but not by a super huge amount. Alternatively, we could give a list of filepaths rather than loading the files, but that would require some odd kiln workarounds.